src/compiler/checker.ts

```diff
@@ -11368,6 +11368,12 @@ module ts {
         function checkSourceFileWorker(node: SourceFile) {
             let links = getNodeLinks(node);
             if (!(links.flags & NodeCheckFlags.TypeChecked)) {
+                links.flags |= NodeCheckFlags.TypeChecked;

```

Comment:
why mark it if we are skipping it? the two checks seem to be more or less the same cost, but it is misleading.